web/app/scripts/src/directives/proportion-field.es.js

```diff
@@ -0,0 +1,55 @@
+import { angular } from '../libraries';
+
+import ProportionField from '../services/proportion-field';
+
+const moduleProperties = {
+    moduleName: 'incuna-surveys.proportion-field-directive'
+};
+
+const module = angular.module(moduleProperties.moduleName, [
+    ProportionField.moduleName
+]);
+
+module.directive('proportionField', [
+    ProportionField.componentName,
+    function (ProportionField) {
+        return {
+            restrict: 'A',
+            scope: {
+                model: '=',
+                options: '=proportionField'
+            },
+            templateUrl: 'templates/incuna-surveys/form/proportion-field.html',
+            link: function (scope) {
+                scope.fields = [];
+                scope.$watch('options', (options) => {
+                    if (options.fieldOptions) {
+                        scope.title = options.fieldOptions.label;
+                    }
+                    if (options.choices) {
+                        scope.fields = ProportionField.buildFields(
+                            options.choices,
+                            options.fieldOptions,
+                            options.autoId
+                        );
+                    }
+                });
+
+                scope.$watch('options.fieldOptions.errors', (errors) => {
+                    if (errors) {
+                        ProportionField.addErrors(scope.fields, errors);
```

Comment:
`addErrors` replaces the `errors` attribute so  clears the errors if there are none - I will add a test